core/tests/performance_framework/selenium_driver.py

```diff
@@ -0,0 +1,221 @@
+    os.path.join(OPPIA_TOOLS_DIR, 'selenium-2.53.2'),
+    os.path.join(OPPIA_TOOLS_DIR, 'xvfbwrapper-0.2.8')
+]
+
+# pylint: disable=import-error, wrong-import-position
+from browsermobproxy import Server
+from selenium import webdriver
+
+# Used for headless testing i.e, making browser windows invisible to the user.
+from xvfbwrapper import Xvfb
+# pylint: enable=import-error, wrong-import-position
+
+CHROME_DRIVER_PATH = os.path.join(
+    '..', 'node_modules', 'protractor', 'selenium', 'chromedriver_2.21')
+
+BROWSERMOB_PROXY_PATH = os.path.join(
+    '..', 'oppia_tools', 'browsermob-proxy-2.1.1', 'bin', 'browsermob-proxy')
+
+
+class SeleniumPerformanceDataFetcher(object):
+    """Fetches performance data for locally served Oppia pages using Selenium
+    and Browsermob-proxy.
+
+    Selenium is used to programitically interact with a browser.
+    Browsermob-proxy is used to capture HTTP Archive(referred to as HAR) data.
+
+    The HTTP Archive format or HAR, is a JSON-formatted archive file format for
+    logging of a web browser's interaction with a site. It is used by a web
+    browser to export detailed performance data about web pages it loads. It
+    contains an entry for every resource required to load and display a
+    web-page. Each entry contains the url requested, request and response
+    headers.
+    Refer: https://dvcs.w3.org/hg/webperf/raw-file/tip/specs/HAR/Overview.html
+    """
+
+    def __init__(self, browser='chrome'):
+        if browser in ['chrome', 'firefox']:
+            self.browser = browser
+        else:
+            raise ValueError
+
+        if self.browser == 'chrome':
+            os.environ["webdriver.chrome.driver"] = CHROME_DRIVER_PATH
+
+        self.chrome_driver_path = CHROME_DRIVER_PATH
+        self.browser_mob_path = BROWSERMOB_PROXY_PATH
+
+        self.server = self.driver = self.proxy = None
+
+    def get_har_dict(self, page_url):
+        """Retrieve HTTP Archive (HAR) or page session stats for a page URL.
+
+        To get it we require the use of a proxy server as we need to record the
+        communication between server and client.
+        """
+        # pylint: disable=unused-variable
+        with Xvfb() as xvfb:
+        # pylint: enable=unused-variable
+            self._setup_proxy()
+            self._setup_driver(use_proxy=True)
+
+            self.proxy.new_har(page_url, options={'captureHeaders': True})
+
+            self.driver.get(page_url)
+
+        har_dict = self.proxy.har
+
+        self._clear_proxy()
+        self._clear_driver()
+
+        return har_dict
+
+    def get_har_dict_cached_state(self, page_url):
+        """Retrieve HTTP Archive (HAR) or page session stats for a page URL
+        while simulating a cached state i.e, a return user.
+
+        To get it we require the use of a proxy server as we need to record the
+        communication between server and client.
+        """
+        # pylint: disable=unused-variable
+        with Xvfb() as xvfb:
```

Comment:
Ah, looking more into the documentation, understand that it starts Xvfb() and stops it automatically.